optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/plugin/routing/AirDistanceRouter.java

```diff
@@ -21,16 +21,18 @@
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 
+import javax.enterprise.context.ApplicationScoped;
+
 import org.optaweb.vehiclerouting.domain.Coordinates;
 import org.optaweb.vehiclerouting.service.distance.DistanceCalculator;
 import org.optaweb.vehiclerouting.service.region.BoundingBox;
 import org.optaweb.vehiclerouting.service.region.Region;
 import org.optaweb.vehiclerouting.service.route.Router;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
-import org.springframework.stereotype.Component;
 
-@Component
-@ConditionalOnProperty(prefix = "app.routing", name = "engine", havingValue = "air")
+import io.quarkus.arc.properties.IfBuildProperty;
+
+@ApplicationScoped
+@IfBuildProperty(name = "app.routing.engine", stringValue = "AIR")
```

Comment:
See `RoutingProperties`. The `app.routing.engine` property type is an enum, which automatically resulted in property validation and auto-completion with Spring Boot. The enum values are uppercase, following the convention. Spring Boot was case-insensitive when evaluating the config property value and accepted both `air` and `AIR` but Quarkus seems to only accept `AIR`.